accepted/2023/preview-apis/preview-apis.md

```diff
@@ -0,0 +1,194 @@
+diagnostic ID. The severity is always error.
+
+The semantics are identical to how obsolete is tracked, except that it will
+always raise a diagnostic regardless of whether the call site is marked as
+preview or not. There is also no special treatment when both caller and callee
+are in the same assembly -- any use will generate a diagnostic and the caller is
+expected to suppress it, with the usual means (i.e. `#pragma` or project-wide
+`NoWarn`).
+
+## Tracking Items
+
+* The overall work of enabling this feature for libraries
+    - [dotnet/runtime#77869](https://github.com/dotnet/runtime/issues/77869)
+* Shipping `RequiresPreviewFeaturesAttribute` for downlevel
+    - [dotnet/runtime#79479](https://github.com/dotnet/runtime/issues/79479)
+* Highlighting preview APIs in docs
+    - [dotnet/dotnet-api-docs#6861](https://github.com/dotnet/dotnet-api-docs/issues/6861)
+* Highlighting preview APIs in IDE
+    - [dotnet/roslyn#65915](https://github.com/dotnet/roslyn/issues/65915)
+
+[preview-features]: ../../2021/preview-features/preview-features.md
+
+## Q & A
+
+### Why didn't we evolve the existing RequiresPreviewFeaturesAttribute?
+
+We considered that initially, but there are major differences in the desired
+user experience for `RequiresPreviewFeaturesAttribute` and library preview
+features:
+
+1. `RequiresPreviewFeaturesAttribute` is designed to be viral. That is, by
+   default any library that turns it on also requires its consumers to turn it
+   on. This makes sense for runtime preview features but not so much for
+   libraries who simply offer some APIs that aren't stable yet.
+
+2. `RequiresPreviewFeaturesAttribute` is designed to be managed via a single
+   MSBuild property, specifically `EnablePreviewFeatures`. This is desirable for
+   library features where wanting to use a preview API in some library doesn't
+   necessarily mean that one wants to use any preview library of any library.
+
+3. `RequiresPreviewFeaturesAttribute` also covers tooling preview features and
+   generally requires the runtime and SDK to match. That is, you cannot use
+   preview features in `net6.0` unless you're using the .NET 6.0 SDK. For
+   example, if you try to use the .NET 7.0 SDK, you'll get a build error telling
+   you that this isn't supported. This alignment isn't necessary for library
+   preview APIs and in fact highly undesirable.
+
+We considered offering a different mode for `RequiresPreviewFeaturesAttribute`
+instead of using a different attribute but we believe this makes it harder to
+understand as these are really disjoint features. Having different names for
+them makes it easier to communicate that difference to our users.
+
+### Why is this not an analyzer?
+
+The compiler team suggested to make this a compiler behavior, rather than an
+analyzer. This would allow the attribute to have a `DiagnosticId` which regular
+analyzers can't (because a given analyzer has to tell the compiler upfront which
+diagnostic IDs it will raise).
```

Comment:
```suggestion
The compiler team suggested to make this a compiler behavior, rather than an
analyzer. This would allow the attribute to specify custom `DiagnosticId` values which regular
analyzers can't (because a given analyzer has to tell the compiler upfront which
diagnostic IDs it will raise).
```